lib/Sema/CSGen.cpp

```diff
@@ -3207,6 +3207,15 @@ namespace {
     }
 
     Type visitForceValueExpr(ForceValueExpr *expr) {
+      // Diagnose force-unwrapping a 'nil' literal
+      auto &DE = CS.getASTContext().Diags;
+      auto subExpr = expr->getSubExpr();
+      if (isa<NilLiteralExpr>(subExpr)) {
+          DE.diagnose(subExpr->getLoc(), diag::cannot_force_unwrap_nil_literal);
+          return Type();
+      }
```

Comment:
What about `Optional<T>.none!`, should that be diagnosed too?

Reply:
I think that's a different problem with explicitly using `nil` literal so we can address that separately.

Reply:
In my solution, if T is 'nil' it will go into the if statement and get diagnosed as cannot_force_unwrap_nil_literal